File: C Sharp/Object Validator/Runtime/And.cs
namespace Quicksilver.ObjectValidator.Runtime;
class And(bool reversed, string? fieldExpression, Condition[] conditions) : Condition(reversed, fieldExpression)
{
    protected readonly Condition[] conditions = conditions;

    protected override bool IsFulfilledBy(object? value, string? fullFieldExpression, HashSet<string> passedFields, HashSet<string> failedFields)
    {
        HashSet<string> newPassedFields = [];
        foreach (Condition condition in conditions)
            if (!condition.Check(value, fullFieldExpression, newPassedFields, failedFields))
                return false;
        passedFields.UnionWith(newPassedFields);
        return true;
    }
}

File: Java/Object Validator/lib/src/main/java/com/quicksilver/objectvalidator/runtime/And.java
package com.quicksilver.objectvalidator.runtime;

import java.util.HashSet;

public class And extends Condition {
    protected final Condition[] conditions;

    public And(boolean reversed, String fieldExpression, Condition... conditions) {
        super(reversed, fieldExpression);
        this.conditions = conditions;
    }

    @Override
    protected boolean isFulfilledBy(Object field, String fullFieldExpression, HashSet<String> passedFields, HashSet<String> failedFields) throws ReflectiveOperationException, IllegalArgumentException, NoSuchFieldException, SecurityException {
        HashSet<String> newPassedFields = new HashSet<>();
        for (Condition condition : conditions)
            if (!condition.check(field, fullFieldExpression, newPassedFields, failedFields))
                return false;
        passedFields.addAll(newPassedFields);
        return true;
    }
}
